### PR TITLE
Shorten test failure error messages

### PR DESCRIPTION
**Pull request by zta6**
Shorten test failure messages to only show first N test failures (25 at the moment), rather than posting extremely long error messages with 150+ test failures. 

Here is an example of the new test error message output with N set to 5
Capture

### CODE REVIEW

File: vars/getFailureMessage.groovy [hunk 1]
```diff
@@ -55,20 +55,28 @@ def formatForJSON(String message) {
 
 def getTestFailures(String log) {
 	def maxStackTraceLength = 500
-	def foundFailures = false
+	def failuresFound = 0
 	def failedTestsMessage = "<h4>Failed Tests:</h4> <ul>"
 	def failedTestPattern = Pattern.compile("[a-z].*<<< (FAILURE|ERROR)!\n")
 	def failedTestMatcher = failedTestPattern.matcher(log)
+	def maxFailureNum = 25
 	while(failedTestMatcher.find()) {
```
Comment (rymach) [inline vars/getFailureMessage.groovy]:
```suggestion
	while(failedTestMatcher.find() && ++failuresFound <= maxFailureNum) {
```

File: vars/getFailureMessage.groovy [hunk 2]
```diff
@@ -55,20 +55,28 @@ def formatForJSON(String message) {
...
 		return ""
 	}
+
+	if(failuresFound > maxFailureNum) {
+		 failedTestsMessage += "...and " + (failuresFound - maxFailureNum) + " more test failures.</ul>"
+	} else {
+		failedTestsMessage += "</ul>"
+	}
 	
 	failedTestsMessage += "\n\n```\n"
```
Comment (rymach) [inline vars/getFailureMessage.groovy]:
```suggestion
	failedTestsMessage += "</ul>\n\n```\n"
```

File: vars/getFailureMessage.groovy [hunk 3]
```diff
@@ -55,20 +55,28 @@ def formatForJSON(String message) {
...
 
-	if (!foundFailures) {
+	if (failuresFound == 0) {
 		return ""
 	}
+
+	if(failuresFound > maxFailureNum) {
+		 failedTestsMessage += "...and " + (failuresFound - maxFailureNum) + " more test failures.</ul>"
+	} else {
+		failedTestsMessage += "</ul>"
```
Comment (rymach) [inline vars/getFailureMessage.groovy]:
```suggestion
```

File: vars/getFailureMessage.groovy (hunk 1 above)
Comment (zta6) [reply vars/getFailureMessage.groovy]: I actually want the loop to run through all the test failures to count them. 
Could commit this suggestion and then modify the regex at L74 to have capturing groups and get the # of test failures that way but this works and its less complicated.

Comment (rymach) [reply vars/getFailureMessage.groovy]: my mistake

**Pull request merged**